omod/src/main/java/org/openmrs/module/smartonfhir/web/servlet/SmartAppSelectorServlet.java

```diff
@@ -15,15 +15,27 @@
 
 import java.io.IOException;
 
+import ca.uhn.fhir.rest.server.IServerAddressStrategy;
 import org.apache.commons.lang3.StringUtils;
 import org.apache.http.HttpStatus;
+import org.springframework.beans.factory.annotation.Autowired;
 
 public class SmartAppSelectorServlet extends HttpServlet {
 	
+	@Autowired
+	private IServerAddressStrategy iServerAddressStrategy;
+	
 	@Override
 	protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+		String baseURL = iServerAddressStrategy.determineServerBase(req.getServletContext(), req);
```

Comment:
@ibacher @VaishSiddharth I get stuck at this point. I want to autowire the `IServerAddressStrategy` to get the base URL implemented in FHIR2 module [here](https://github.com/openmrs/openmrs-module-fhir2/blob/e04d9692554840652dddd32ab6ee3756f43ef87d/omod/src/main/java/org/openmrs/module/fhir2/web/util/OpenmrsFhirAddressStrategy.java#L36) but getting the `NullPointerException` in this line. Can you please review why I'm not able to autowire this external component?

Reply:
@ibacher If I am not wrong we can only use classes that are exposed by the API module and not those which are present in omod. Am I right?

Reply:
You can only rely on `@Autowire` working in situations where the object is created by Spring. In this case, the `SmartAppSelectorService` isn't, it's created [by this line](https://github.com/openmrs/openmrs-core/blob/71209827fec07e9c92480ab0af95d7b91b83d315/web/src/main/java/org/openmrs/module/web/WebModuleUtil.java#L455), so this will never be a Spring-managed bean, which means it will never autowire correctly.

To do something like this, you'd need to something like:

```
org.openmrs.api.context.Context.getRegisteredComponent("openmrsFhirAddressStrategy", IServerAddressStrategy.class);
```

Reply:
Thank you, I will change this.